tests/test_data_iterator.py

```diff
@@ -0,0 +1,123 @@
+from re import L
+from typing import Any, Dict, List, Optional
+
+import pandas as pd
+import pandas.testing as pd_test
+import pytest
+
+from packages.python.accern_data import create_data_client, DT_FORMAT
+from packages.python.accern_data.util import (
+    DEFAULT_CHUNK_SIZE,
+    EXAMPLE_URL,
+    load_json,
+)
+
+DEFAULT_CHUNK_SIZE_LIST = [
+    1, 5, 8, 3, 57, 13, 88, 86, 18, 9, 91, 68, 1, 99, 1, 22, 1, 12, 3, 8, 16,
+    2, 68, 1, 71, 2, 81, 3, 89, 69, 1, 96, 97, 93, 17, 1, 98, 92, 97, 14, 90,
+    48, 1, 99, 98, 32, 7, 95, 96, 14, 1, 98, 98, 98, 34, 5, 98, 40, 3,
+]
+
+
+@pytest.mark.parametrize("chunk_size", [1, 5, 1000, None])
+def test_csv_full_iterator(chunk_size: Optional[int]) -> None:
+    start_date = "2022-01-03"
+    end_date = "2022-03-04"
+    client = create_data_client(EXAMPLE_URL, "SomeRandomToken")
+    client.set_mode("csv", split_dates=False, chunk_size=chunk_size)
+    dataframe = pd.read_csv("tests/data/data-2022.csv")
+    n_full_chunks = None
+    if chunk_size is not None:
+        n_full_chunks = dataframe.shape[0] // chunk_size
+    df_lengths = []
+    dfs: List[pd.DataFrame] = []
+    for df in client.iterate_range(start_date=start_date, end_date=end_date):
+        df_lengths.append(df.shape[0])
+        dfs.append(df)
+    concat_df = pd.concat(dfs)
+    assert (~concat_df.duplicated()).all(), "Duplicate entry is present."
+    if chunk_size is not None and n_full_chunks is not None:
+        for idx in range(n_full_chunks):
+            assert df_lengths[idx] == chunk_size
+        if dataframe.shape[0] % chunk_size != 0:
+            assert df_lengths[-1] == dataframe.shape[0] % chunk_size
+    else:
+        for idx, df in enumerate(dfs):
+            assert df.shape[0] == DEFAULT_CHUNK_SIZE_LIST[idx]
```

Comment:
this cannot be the same for split dates and not split dates

Reply:
yes it is. In default chunk_size case, we yield data as we get from api, so that would be same for csv_full or csv_date.